symfony/framework-bundle/3.3/src/Kernel.php

```diff
@@ -35,6 +36,27 @@ public function registerBundles()
         }
     }
 
+    public static function bootstrapEnvironment(string $env = null)
```

Comment:
I think `$env` should be null here.
You already pass not null value from `bin/console`. You can pass a same value from `public/index.php` (you call `bootstrapEnvironment` and create `$env` var on the next line, just swap that lines and pass `$env` to `bootstrapEnvironment`!). Also you can pass `$_SERVER['APP_ENV'] ?? 'test'` from `tests/bootstrap.php`.

Reply:
@weaverryan environment can be not set if `'test' === $_SERVER['APP_ENV']`. And if so, then `.env` file(s) should be loaded (or `.env.test`?). And if so, then `$env` should be passed from `tests/bootstrap.php` too

Reply:
if $_SERVER['APP_ENV'] was defined, we assume we already have the enviromnents loaded.